drafts/bdd-specflow.md

```diff
@@ -0,0 +1,191 @@
+---
+title: How to use Behaviour-Driven Development (BDD) with C# and SpecFlow
+subtitle: Here is an introductory guide on how to use Behaviour-Driven Development (BDD) with C# and SpecFlow in a .NET application.
+domain: software-engineering-corner.hashnode.dev
+tags: web-development, azure, build-tool, sass
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1682689039164/_2bkGoHwv.webp?auto=format
+publishAs: alexander-duschek
+hideFromHashnodeCommunity: false
+ignorePost: true
+---
+
+Behaviour-Driven Development (BDD) is a technique of agile software development to ensure that the requirements of the software to be developed are fulfilled. To achieve this a special syntax called Gherkin (see below) is used to write down the specification which then can be translated by tools into test case skeletons. The developers add the test code and write/enhance the application until the test passes.
+A big advantage is that stakeholders, customers, business analysts, requirement engineers, all parties of a project can provide tests.
+
+## BDD Key features
+
+- Everyone can provide test cases
+- Test cases are easy to read
+- Good tool support for automation
+- Developers need less domain knowledge
+- Developers can focus on writing tests and features instead of finding test cases
+- Project documentation on the fly
+
+## What about Test-Driven Development (TDD)?
+
+Simply put, if a software developer uses the TDD technique, he first creates unit tests based on the requirements or the specification of the software or feature. Afterward, he enhances the software until the tests will eventually pass. For more details read the following article.
+
+https://medium.com/geekculture/test-driven-development-that-every-developer-should-know-about-with-example-c5f68af547a6
```

Comment:
I would put this link directly into the text like `[This article on Medium](https...) explains TDD in more detail`.